control-plane/csi-driver/src/client.rs

```diff
@@ -483,4 +463,42 @@ impl IoEngineApiClient {
 
         Ok(snapshot.into_body())
     }
+
+    /// Register a frontend node.
+    pub async fn register_app_node(
```

Comment:
I'd say leave this as is with pub(crate) in the csi-controller and add a new file for csi-node with just the bits we need - ie app-node stuff

Reply:
Not sure if we can do that, we would have to duplicate the RestApiClient structure at both places then.

Reply:
Done.